dss/api/bundles/__init__.py

```diff
@@ -125,7 +125,18 @@ def put(uuid: str, replica: str, json_request_body: dict, version: str):
     bundle_manifest_key = BundleFQID(uuid=uuid, version=version).to_key()
 
     # decode the list of files.
-    files = [{'user_supplied_metadata': file} for file in json_request_body['files']]
+    files = list()
+    filenames: typing.Set[str] = set()
+    for file in json_request_body['files']:
+        name = file['name']
+        if name in filenames:
+            raise DSSException(
+                requests.codes.bad_request,
+                "duplicate_filename",
+                "A bundle cannot contain duplicate filenames"
```

Comment:
Consider including the offending file name in the error message.

Also consider determining all duplicates before raising the exception and including all duplicates in the error message.

Reply:
Filename included in error message.

In lieue of extra duplication-detecting logic, a warning has been added indicating further  duplicate filename may be present in the bundle.

Maybe it wouldn't be a lot of extra logic, but, in general, it seems reasonable for our API backend to fail as quickly as possible on validation errors.